### PR TITLE
for Volumes

### PR DESCRIPTION
**Pull request by KostyaSha**
Filters dirty inputs more carefully

### CODE REVIEW

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java [hunk 1]
```diff
@@ -110,16 +111,32 @@ public DockerTemplateBase(String image,
...
+        return list.toArray(new String[list.size()]);
+    }
+
+    //TODO move/replace
+    public static String[] filterStringArray(String[] arr) {
+        final ArrayList<String> strings = new ArrayList<>();
+        if (arr != null) {
+            for (String s : arr) {
+                if (StringUtils.isNotEmpty(s)) {
+                    strings.add(s.trim());
```
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: here can be npe

**Comment by KostyaSha** [comment]
@lanwen are you ok with this changes?

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java [hunk 2]
```diff
@@ -110,16 +111,32 @@ public DockerTemplateBase(String image,
 
     protected Object readResolve() {
         if (volumesFrom != null) {
```
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: just check once
```
if (StringUtils.isNotBlank(volumesFrom)) {
                setVolumesFrom2(new String[]{volumesFrom});
}
```

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java [hunk 3]
```diff
@@ -110,16 +111,32 @@ public DockerTemplateBase(String image,
...
+    //TODO move/replace
     public static String[] splitAndFilterEmpty(String s, String separator) {
         List<String> list = Splitter.on(separator).omitEmptyStrings().splitToList(s);
-        return list.toArray(new String[0]);
+        return list.toArray(new String[list.size()]);
+    }
+
+    //TODO move/replace
+    public static String[] filterStringArray(String[] arr) {
+        final ArrayList<String> strings = new ArrayList<>();
```
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: don't use concrete class names if it has interface - in this case it should be List

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java (hunk 2 above)
Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: You are wrong, this code lines not about beauty, it data migration.

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java (hunk 3 above)
Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: It filters array of strings.

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java (hunk 2 above)
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: Are you kidding me?

What for you need two ifs if second if also checks first?
And why you asking me ok, or not, if you don't listen me?

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java (hunk 3 above)
Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: As you see i mentioned that it need to be removed or replaced in future. I have no time for beatifying code that will be removed soon.

**Comment by lanwen** [comment]
ok to me

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java (hunk 2 above)
Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: Your proposed replacement absolutely wrong from data migration view point. Even if current code duplicated, then it not an issue. And as i repeated 100 times personally to you "configuration handling will be changed" no sense in wasting time on code beauty. Instead of continuing unbundling launcher i spent the whole day on unrelated to fixes comments.

Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java]: So you can just write `will be fixed next time` and merge without any doubt. And next time don't write duplicated code.
